File: java/L1500/LeetCode1518.java
package L1500;

public class LeetCode1518 {
    public int numWaterBottles(int numBottles, int numExchange) {
        int sum = numBottles;
        int noWater = 0;
        while (numBottles != 0) {
            int t = (numBottles + noWater) / numExchange;
            sum += t;
            noWater = (numBottles + noWater) % numExchange;
            numBottles = t;
        }
        return sum;
    }
}

File: cpp/L1500/LeetCode1518.cc
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int ans{numBottles};
        for (int cur{0};numBottles + cur >= numExchange;) {
            int bottle = (numBottles + cur) / numExchange;
            cur = (numBottles + cur) % numExchange;
            ans += bottle;
            numBottles = bottle;
        }      
        return ans;
    }
};